pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -735,27 +794,62 @@ installPhase:
 
 		default:
 			for _, installWorker := range pplnPhase.q {
+				if enterpriseApi.ScopeLocal != getAppSrcScope(installWorker.afwConfig, installWorker.appSrcName) {
+					scopedLog.Error(nil, "Install worker with wrong scope", "name", installWorker.cr.GetName(), "namespace", installWorker.cr.GetNamespace(), "pod name", installWorker.targetPodName, "App name", installWorker.appDeployInfo.AppName, "digest", installWorker.appDeployInfo.ObjectHash)
+					continue
+				}
+
 				phaseInfo := getPhaseInfoByPhaseType(installWorker, enterpriseApi.PhaseInstall)
 				if checkIfWorkerIsEligibleForRun(installWorker, phaseInfo, enterpriseApi.AppPkgInstallComplete) {
 					installWorker.waiter = &pplnPhase.workerWaiter
 					select {
 					case pplnPhase.msgChannel <- installWorker:
 						scopedLog.Info("Install worker got a run slot", "name", installWorker.cr.GetName(), "namespace", installWorker.cr.GetNamespace(), "pod name", installWorker.targetPodName, "App name", installWorker.appDeployInfo.AppName, "digest", installWorker.appDeployInfo.ObjectHash)
 						installWorker.isActive = true
+
+						// ToDo Start: sgontla: Fix the flow once CSPL-1449 is in place
+						var localInstallCtxt *localScopeInstallContext = &localScopeInstallContext{
+							worker: installWorker,
+							//sem:
+						}
+
+						installWorker.waiter.Add(1)
```

Comment:
so why this? I don't see `runPlaybook` as a separate goroutine here and if it is not, it is synchronous call anyways, right? or unless I am missing something basic here :)